### PR TITLE
New TestSensor backend

### LINKED ISSUE #95: TestSensor could have a feedback backend
While the FeedbackController has yet to be used in production, we could add a simulated process into the TestSensorServer to allow us to test feedback performance without needing hardware.

### PR DESCRIPTION
**Pull request by darrylmasson**
It's a relatively rudimentary simulation (the things I plan on adding...), but it should be fully-featured-enough now to test new things we're adding to the Experimental branch. The TestSensorServer is now much cleaner and easier to maintain, and the TestDatabase probably won't survive because it's unfinished and I don't feel like continuing with it.
Closes #95

### CODE REVIEW

File: tests/TestSensor.py [hunk 1]
```diff
@@ -6,10 +6,9 @@ class TestSensor(Doberman.LANSensor):
...
-        self.reading_pattern = re.compile(('OK;(?P<value>%s)' %
-            Doberman.utils.number_regex).encode())
+        self.reading_commands = {'t_in' : 'READ:t_in',
+                                 't_amb' : 'READ:t_amb',
+                                 'q_add': 'READ:q_add'}
+        self.reading_pattern = re.compile(f'OK;(?P<value>{Doberman.utils.number_regex})'.encode())
         self.command_patterns = [
-                (re.compile('d(?P<sides>[1-9][0-9]*)'),
-                    lambda m : f'SET:sides={m.group("sides")}')]
+            (re.compile(f'SET q_add=({Doberman.utils.number_regex})', lambda m : f'SET:q_add={m.group(1)}'))]
```
Comment (jarongrigat) [inline tests/TestSensor.py:14]: The parantheses are wrong, I think. Shouldnt it be:
` (re.compile(f'SET q_add=({Doberman.utils.number_regex})'), lambda m : f'SET:q_add={m.group(1)}')]`

**Review by jarongrigat — CHANGES_REQUESTED** [review]
Looks good. I wonder what the shitstorm will look like :)
Also I don't quite understand the TestDatabase. Can't we just have a test db in our standard mongoDB?

File: tests/TestSensor.py (hunk 1 above)
Comment (darrylmasson) [reply tests/TestSensor.py]: Ideally yes but I'll make it compatible with #117

**Comment by darrylmasson** [comment]
The TestDatabase was originally intended to allow one to test without a connection to the actual database, but it's much less hassle to ssh port-forward. We do have a "testing" experiment in the Mongo that things should look at.

**Review by jarongrigat — APPROVED** [review]

**Pull request merged**